### PR TITLE
Release automation for 0.18

### PR DESCRIPTION
**Pull request by rossabaker**
Most of this is old news, but it's another non-trivial merge.

### CODE REVIEW

File: blaze-server/src/main/scala/org/http4s/server/blaze/Http2NodeStage.scala [hunk 1]
```diff
@@ -185,20 +185,23 @@ private class Http2NodeStage[F[_]](
...
           }
-          .map(renderResponse(_))
-        catch serviceErrorHandler(req).andThen(_.widen[MaybeResponse[F]])
-      } {
-        case Right(_) =>
-          IO.unit
-        case Left(t) =>
-          IO(logger.error(t)("Error rendering response"))
-      }
+      })
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/Http2NodeStage.scala]: This is a translation.

File: core/src/main/scala/org/http4s/Message.scala [hunk 2]
```diff
@@ -366,10 +366,14 @@ sealed trait MaybeResponse[F[_]] {
 object MaybeResponse extends MaybeResponseInstances
 
 trait MaybeResponseInstances {
-  implicit def http4sMonoidForMaybeResponse[F[_]]: Monoid[MaybeResponse[F]] =
+
+  /** Intentionally not implicit. Do not use unless you've read and understood the
+    * deprecation warning on [[instance]].
+    */
+  def http4sMonoidForMaybeResponse[F[_]]: Monoid[MaybeResponse[F]] =
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/Message.scala]: I didn't leave behind a deprecated implicit version.  This all becomes obsolete with #1424.

And now I see that the scaladoc is a lie.

File: core/src/main/scala/org/http4s/util/io/package.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  def captureWriter(f: JWriter => Unit)(implicit S: Strategy): Process[Task, ByteVector] =
+    captureOutputStream(osCallbackToWriterCallback(f))
+
+  private def osCallbackToWriterCallback(f: JWriter => Unit) = { out: OutputStream =>
+    val w = new OutputStreamWriter(out, StandardCharsets.UTF_8)
+    f(w)
+    w.flush()
+  }
+}
+  */
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/util/io/package.scala]: I didn't mean to exhume this.

File: project/Http4sPlugin.scala [hunk 4]
```diff
@@ -56,7 +92,90 @@ object Http4sPlugin extends AutoPlugin {
...
+      val master = http4sMasterBranch.value
+
+      Seq(
+        checkSnapshotDependencies.when(release),
+        inquireVersions.when(release),
+        setReleaseVersion.when(release),
+        tagRelease.when(release),
+        runTestWithCoverage,
+        releaseStepCommand("mimaReportBinaryIssues"),
+        releaseStepCommand("test:scalafmt::test").when(primary),
```
Comment (rossabaker) [inline project/Http4sPlugin.scala]: This is new in 0.18.  It reflects the old `validate`.

File: project/Http4sPlugin.scala [hunk 5]
```diff
@@ -56,7 +92,90 @@ object Http4sPlugin extends AutoPlugin {
...
+        releaseStepCommand("docs/makeSite").when(primary),
+        releaseStepCommand("website/makeSite").when(primary),
+        openSonatypeRepo.when(publish && release),
+        publishToSonatypeWithoutInstrumentation.when(publish),
```
Comment (rossabaker) [inline project/Http4sPlugin.scala]: This is a renamed target in rig-5.

File: project/plugins.sbt [hunk 6]
```diff
@@ -9,7 +9,7 @@ addSbtPlugin("com.typesafe.sbt"   %  "sbt-twirl"             % "1.3.7")
 addSbtPlugin("io.gatling"         %  "gatling-sbt"           % "2.2.2")
 addSbtPlugin("io.get-coursier"    %  "sbt-coursier"          % "1.0.0-RC11")
 addSbtPlugin("io.spray"           %  "sbt-revolver"          % "0.9.0")
-addSbtPlugin("io.verizon.build"   %  "sbt-rig"               % "5.0.37")
+addSbtPlugin("io.verizon.build"   %  "sbt-rig"               % "5.0.38")
```
Comment (rossabaker) [inline project/plugins.sbt]: .38 gets rid of `-Xcheckinit` on a final publish, which adds runtime drag.

File: server/src/main/scala/org/http4s/server/staticcontent/WebjarService.scala [hunk 7]
```diff
@@ -52,7 +52,7 @@ object WebjarService {
     * @param config The configuration for this service
     * @return The HttpService
     */
-  def apply[F[_]: Monad: Sync](config: Config[F]): HttpService[F] = Service {
+  def apply[F[_]: Monad: Sync](config: Config[F]): HttpService[F] = HttpService.lift {
```
Comment (rossabaker) [inline server/src/main/scala/org/http4s/server/staticcontent/WebjarService.scala]: I'm going to advocate for deprecating `HttpService.lift` in #1424.

Comment (aeons) [reply server/src/main/scala/org/http4s/server/staticcontent/WebjarService.scala]: I support that. So we will have `apply(pf: PartialFunction[Request[F], F[Response[F]]])` and `liftF(f: Request[F] => OptionT[F, Response[F]])` as the ways to create an `HttpService[F]`, right?

**Pull request merged**